xwiki-commons-core/xwiki-commons-legacy/xwiki-commons-legacy-extension/xwiki-commons-legacy-extension-api/src/main/java/org/xwiki/extension/job/history/internal/LegacyExtensionJobHistoryRecorder.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.extension.job.history.internal;
+
+import javax.inject.Named;
+import javax.inject.Singleton;
+
+import org.xwiki.component.annotation.Component;
+import org.xwiki.extension.job.history.ReplayJobStatus;
+import org.xwiki.job.Job;
+import org.xwiki.job.event.JobStartedEvent;
+import org.xwiki.job.internal.AbstractJobStatus;
+import org.xwiki.observation.event.Event;
+
+/**
+ * Legacy implementation of {@link ExtensionJobHistoryRecorder}.
+ *
+ * @version $Id$
+ * @since 15.7RC1
+ */
+@Component
+@Named(ExtensionJobHistoryRecorder.NAME)
+@Singleton
+@Deprecated
+public class LegacyExtensionJobHistoryRecorder extends ExtensionJobHistoryRecorder
+{
+    @Override
+    public void onEvent(Event event, Object source, Object data)
+    {
+        boolean ignore = false;
+        if (event instanceof JobStartedEvent) {
+            Job job = (Job) source;
+            if (job.getStatus() instanceof org.xwiki.job.AbstractJobStatus
```

Comment:
I might have used an aspect for injecting this condition check, but it felt easier to do it like that.